pkg/continuousprofiling/profiler.go

```diff
@@ -0,0 +1,125 @@
+package continuousprofiling
+
+import (
+	"runtime"
+
+	"github.com/grafana/pyroscope-go"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	ErrApplicationName           = errors.New("the ApplicationName must be defined")
+	ErrServerAddress             = errors.New("the ServerAddress must be defined")
+	ErrAtLeastOneProfileIsNeeded = errors.New("at least one profile is needed")
+
+	log = logging.LoggerForModule()
+
+	DefaultProfiles = []pyroscope.ProfileType{
+		pyroscope.ProfileCPU,
+		pyroscope.ProfileAllocObjects,
+		pyroscope.ProfileAllocSpace,
+		pyroscope.ProfileInuseObjects,
+		pyroscope.ProfileInuseSpace,
+		pyroscope.ProfileGoroutines,
+		pyroscope.ProfileMutexCount,
+		pyroscope.ProfileMutexDuration,
+		pyroscope.ProfileBlockCount,
+		pyroscope.ProfileBlockDuration,
+	}
+)
+
+// DefaultConfig creates a new configuration with default properties.
+func DefaultConfig() *pyroscope.Config {
+	return &pyroscope.Config{
+		ApplicationName:   "AppName",
+		ServerAddress:     env.ContinuousProfilingServerAddress.Setting(),
```

Comment:
Do we need any URL formatting here? For example adding or removing `https://` and trailing slashes.

Reply:
Good point! I added some checks in the last commit.